### DISCUSSION TITLE
Haystack Integration for Marqo DocumentStore?

### DISCUSSION THREAD

**Discussion by TuanaCelik** [marqo-ai/marqo#543] [Ideas]
Hey marqo team 👋
I'm on the team that works on [haystack](https://github.com/deepset-ai/haystack), an NLP framework to build with LLMs, implement retrieval, QA, document search etc.
I wanted to post here and see what you would think about the idea of developing a MarqoDocumenStore for the Haystack community? We've just been in an office hour on our Discord and had a community member talk about how they've been enjoying using Marqo as their db of choice. I'm thinking something similar to this by Qdrant: https://haystack.deepset.ai/integrations/qdrant-document-store

**Comment by OwenElliottDev** [comment]
Hi @TuanaCelik, it'd be great to do a MarqoDocumentStore integration with Haystack! We can get started on this in the next couple of weeks.

I had a quick look at the base class and I think we should be able to fit in well with the existing architecture. Looking forward to becoming a part of the Haystack ecosystem.

**Comment by TuanaCelik** [comment]
Great to hear @OwenPendrighElliott - Ideally, you could create something similar to what [Qdrant](https://github.com/qdrant/qdrant-haystack) did and provide it as a pypi package to people, then, I'll create you a Haystack integration page with instructions for people to install it. I'm working on a guideline doc to help you with this, give some hints for testing etc...
Feel free to reach out to me if you like, I'm on your Slack now too ⭐️

**Reply by OwenElliottDev** [reply]
Cheers, I'll take a look at the Qdrant one and go from there. Any documentation or best practices you have for the integrations are welcome! I'll reach out on Slack if we run into any troubles.
